include/openssl/store.h

```diff
@@ -144,6 +145,8 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bio, const char *scheme,
                                   OSSL_STORE_post_process_info_fn post_process,
                                   void *post_process_data);
 
+const char *OSSL_file_stat(const char *uri, struct stat *st);
```

Comment:
Do we really want this as a new public API function? I'd prefer not, if possible.

Reply:
At first I aimed at adding this as an internal function,
but this proved difficult because the function is needed by the provider implementation in `providers/implementations/storemgmt/file_store.c` and by `engines/e_loader_attic.c b/engines/e_loader_attic.c`.

Reply:
For engines I think we can make a copy as they are to be removed in 4.0.

Reply:
If the function was in a separate source file, it could be statically linked to engines. And I do not think we currently create any .so provider from file_store.c.